test/parallel/test-net-listen-after-destroying-stdin.js

```diff
@@ -8,15 +8,15 @@ const net = require('net');
 
 process.stdin.destroy();
 
-const server = net.createServer(common.mustCall(function(socket) {
-  console.log('accepted');
-  socket.end();
-  server.close();
+const server = net.createServer(common.mustCall((socket) => {
+  console.log('accepted...');
+  socket.end(common.mustCall(() => { console.log('finished...'); }));
+  server.close(common.mustCall(() => { console.log('closed'); }));
```

Comment:
For the QUIC tests, I've started using the pattern of creating a `debuglog` for the tests attached to the `test` category, e.g.

```
$ NODE_DEBUG=test ./node test/parallel/test-whatever
```

Then in the tests using

```
const { debuglog } = require('util');
const debug = debuglog('test');
// ...
debug('some message');
```

This works very well to get the appropriate debug output, and very clearly separates purely debug statements from stuff that's part of the test itself. I'd much prefer that pattern over using `console.log()`

It's not just for style either, I have run into cases before where the use of console.log ends up altering the timing of the test just enough to impact the result. It's definitely not common but it can certainly happen.

Reply:
This won't be as effective in cases like this where the test seems to fail only in CI and very infrequently. But yeah, it's preferable. This test already has some console.log() statements in it so we'd have to move it to this new pattern.

> It's not just for style either, I have run into cases before where the use of console.log ends up altering the timing of the test just enough to impact the result. It's definitely not common but it can certainly happen.

Yeah, I've definitely come across that too and that's the most compelling (to me) argument for avoiding `console` in tests.

Also, once the cause of test failures is found, it's often possible to find another way to have surfaced it that wouldn't have involved `console`.

Lots to consider here and a lot of tests to refactor if we move to a new pattern.